frontend/src/components/homepage/homepage.tsx

```diff
@@ -13,24 +14,42 @@ const Homepage: React.FC = () => {
   const dispatch = useDispatch();
   const hasPodcasts = Boolean(podcasts.length);
 
+  const [filter, setFilter] = useState<Array<Podcast>>(podcasts);
+
   useEffect(() => {
     dispatch(homepageActions.loadPodcasts());
   }, []);
 
+  const handleChange = (searchValue: SearchPayload): void => {
+    const { search } = searchValue;
+    const filtered = filterPodcasts(search);
+    const payload = !search ? podcasts : filtered;
+    setFilter(payload);
+  };
+
+  const filterPodcasts = (search: string): Array<Podcast> => {
```

Comment:
Does this filter feature work only on front-end?

Reply:
Yes, I implemented it only on the front-end

Reply:
I needed to make the already filtered data come from the server to the frontend?